NitroxServer/Communication/Packets/Processors/BuildingProcessor.cs

```diff
@@ -1,19 +1,24 @@
+using NitroxModel.DataStructures;
+using NitroxModel.DataStructures.GameLogic;
 using NitroxModel.Packets;
 using NitroxServer.Communication.Packets.Processors.Abstract;
 using NitroxServer.GameLogic;
 using NitroxServer.GameLogic.Bases;
+using NitroxServer.GameLogic.Entities;
 
 namespace NitroxServer.Communication.Packets.Processors;
 
 public abstract class BuildingProcessor<T> : AuthenticatedPacketProcessor<T> where T : Packet
 {
     internal readonly BuildingManager buildingManager;
     internal readonly PlayerManager playerManager;
+    internal readonly EntitySimulation entitySimulation;
 
-    public BuildingProcessor(BuildingManager buildingManager, PlayerManager playerManager)
+    public BuildingProcessor(BuildingManager buildingManager, PlayerManager playerManager, EntitySimulation entitySimulation = null)
```

Comment:
Why `= null`? If it's required, add `Validate.NotNull(entitySimulation);` in the `ClaimBuildPiece` method.

Reply:
It's null because many processors don't need this part but it's just bad design at this point so I'll just add it properly to all classes requiring this instance

Reply:
The full constructor (with entitySimulation) should only be used when the processor needs to use ClaimBuildPiece but it'd need to make a 2nd constructor for the ones which don't need to use ClaimBuildPiece. So I made the 2 constructors into one.